app/models/agents/evernote_agent.rb

```diff
@@ -0,0 +1,377 @@
+            "sourceURL": "...",
+            "resources" : [
+              {
+                "url": "resource1_url",
+                "name": "resource1_name",
+                "mime_type": "resource1_mime_type"
+              }
+              ...
+            ]
+          }
+    MD
+
+    default_schedule "never"
+
+    def working?
+      event_created_within?(interpolated['expected_update_period_in_days']) && !recent_error_logs?
+    end
+
+    def default_options
+      {
+        "expected_update_period_in_days" => "2",
+        "mode" => "update",
+        "include_xhtml_content" => "false",
+        "note" => {
+          "title" => "{{title}}",
+          "content" => "{{content}}",
+          "notebook" => "{{notebook}}",
+          "tagNames" => "{{tag1}}, {{tag2}}"
+        }
+      }
+    end
+
+    def validate_options
+      errors.add(:base, "mode must be 'update' or 'read'") unless %w(read update).include?(options[:mode])
+
+      if options[:mode] == "update" && schedule != "never"
+        errors.add(:base, "when mode is set to 'update', schedule must be 'never'")
+      end
+
+      if options[:mode] == "read" && schedule == "never"
+        errors.add(:base, "when mode is set to 'read', agent must have a schedule")
+      end
+
+      errors.add(:base, "expected_update_period_in_days is required") unless options['expected_update_period_in_days'].present?
+
+      if options[:mode] == "update" && options[:note].values.all?(&:empty?)
+        errors.add(:base, "you must specify at least one note parameter to create or update a note")
+      end
+    end
+
+    def include_xhtml_content?
+      options[:include_xhtml_content] == "true"
+    end
+
+    def receive(incoming_events)
+      if options[:mode] == "update"
+        incoming_events.each do |event|
+          note = note_store.create_or_update_note(note_params(event))
+          create_event :payload => note.attr(include_content: include_xhtml_content?)
+        end
+      end
+    end
+
+    def check
+      if options[:mode] == "read"
+        opts = note_params(options)
+
+        # convert time to evernote timestamp format:
+        # https://dev.evernote.com/doc/reference/Types.html#Typedef_Timestamp
+        opts.merge!(agent_created_at: created_at.to_i * 1000)
+        opts.merge!(last_checked_at: (memory[:last_checked_at] ||= created_at.to_i * 1000))
+
+        if opts[:tagNames]
+          opts.merge!(notes_with_tags: (memory[:notes_with_tags] ||=
+            NoteStore::Search.new(note_store, {tagNames: opts[:tagNames]}).note_guids))
+        end
+
+        notes = NoteStore::Search.new(note_store, opts).notes
+        notes.each do |note|
+          memory[:notes_with_tags] << note.guid unless memory[:notes_with_tags].include?(note.guid)
+
+          create_event :payload => note.attr(include_resources: true, include_content: include_xhtml_content?)
+        end
+
+        memory[:last_checked_at] = Time.now.to_i * 1000
+        save!
```

Comment:
This shouldn't be needed.  The Agent will be saved for you after every check if memory has changed.  (Possibly even if it hasn't...)